src/topology/bounded_continuous_function.lean

```diff
@@ -495,4 +495,90 @@ instance : normed_space 𝕜 (α →ᵇ β) := ⟨λ c f, norm_of_normed_group_l
+
+variables [topological_space α]
+
+instance {R : Type*} [normed_ring R] : ring (α →ᵇ R) :=
+{ one := const α 1,
+  mul := λ f g, of_normed_group (f * g) (f.2.1.mul g.2.1) (∥f∥ * ∥g∥) $ λ x,
+    le_trans (normed_ring.norm_mul (f x) (g x)) $
+      mul_le_mul (f.norm_coe_le_norm x) (g.norm_coe_le_norm x) (norm_nonneg _) (norm_nonneg _),
+  one_mul := λ f, ext $ λ x, one_mul (f x),
+  mul_one := λ f, ext $ λ x, mul_one (f x),
+  mul_assoc := λ f₁ f₂ f₃, ext $ λ x, mul_assoc _ _ _,
+  left_distrib := λ f₁ f₂ f₃, ext $ λ x, left_distrib _ _ _,
+  right_distrib := λ f₁ f₂ f₃, ext $ λ x, right_distrib _ _ _,
+  .. bounded_continuous_function.add_comm_group }
+
+instance {R : Type*} [normed_ring R] : normed_ring (α →ᵇ R) :=
+{ norm_mul := λ f g, norm_of_normed_group_le _ (mul_nonneg (norm_nonneg _) (norm_nonneg _)) _,
+  .. bounded_continuous_function.normed_group }
+
+end normed_ring
+
+section normed_algebra
+/- In this section, if `γ` is a normed algebra, then we show that the space of bounded
+continuous functions from `α` to `γ` inherits a normed algebra structure, by using
+pointwise operations and checking that they are compatible with the uniform distance. -/
+
+variables {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
+variables [topological_space α] [normed_group β] [normed_space 𝕜 β]
+variables [normed_ring γ] [normed_algebra 𝕜 γ]
+variables {f g : α →ᵇ γ} {x : α} {c : 𝕜}
+
+def C : ring_hom 𝕜 (α →ᵇ γ) :=
+{ to_fun    := λ (c : 𝕜), const α (algebra.to_ring_hom.to_fun c),
+  map_one'  := by rw algebra.to_ring_hom.map_one'; exact rfl,
+  map_mul'  := λ c₁ c₂, by rw algebra.to_ring_hom.map_mul'; exact rfl,
+  map_zero' := by rw algebra.to_ring_hom.map_zero'; exact rfl,
+  map_add'  := λ c₁ c₂, by rw algebra.to_ring_hom.map_add'; exact rfl }
+
+instance : algebra 𝕜 (α →ᵇ γ) :=
+{ to_ring_hom := C,
+  commutes' := λ c f, by ext; apply algebra.commutes',
+  smul_def' := λ c f, by ext; apply algebra.smul_def',
+  ..bounded_continuous_function.module,
+  ..bounded_continuous_function.ring }
+
+instance [nonempty α] : normed_algebra 𝕜 (α →ᵇ γ) :=
+{ norm_algebra_map_eq := λ c, begin
+    calc ∥ (algebra_map 𝕜 (α →ᵇ γ)).to_fun c∥ = ∥(algebra_map 𝕜 γ) c∥ : _
+    ... = ∥c∥ : norm_algebra_map_eq _ _,
+    apply norm_const_eq ((algebra_map 𝕜 γ) c), assumption,
+  end,
+  ..bounded_continuous_function.algebra }
+
+/- If `β` is a normed 𝕜-space, then we show that the space of bounded continuous
+functions from `α` to `β` is naturally a module over the algebra of bounded continuous
+functions from `α` to 𝕜. -/
+
+instance has_scalar' : has_scalar (α →ᵇ 𝕜) (α →ᵇ β) :=
+⟨λ (f : α →ᵇ 𝕜) (g : α →ᵇ β), of_normed_group (λ x, (f x) • (g x))
+(continuous.smul f.2.1 g.2.1) (∥f∥ * ∥g∥) (λ x, begin
+  calc ∥f x • g x∥ ≤ ∥f x∥ * ∥g x∥ : normed_space.norm_smul_le _ _
+  ... ≤ ∥f∥ * ∥g x∥ : mul_le_mul_of_nonneg_right (f.norm_coe_le_norm _) (norm_nonneg _)
+  ... ≤ ∥f∥ * ∥g∥ : mul_le_mul_of_nonneg_left (g.norm_coe_le_norm _) (norm_nonneg _),
```

Comment:
You can use `mul_le_mul` to change both terms simultaneously.

Reply:
Fixed in 8bc2f82